modules/tinymce/src/plugins/searchreplace/main/ts/core/Actions.ts

```diff
@@ -183,19 +175,19 @@ const replace = (editor: Editor, currentSearchState: Cell<SearchState>, text: st
   const node = editor.getBody();
   const nodes = Tools.grep(Tools.toArray(node.getElementsByTagName('span')), isMatchSpan);
   for (let i = 0; i < nodes.length; i++) {
-    const nodeIndex = getElmIndex(nodes[i]);
+    const nodeIndex = getElmIndex(nodes[i]) as string;
 
     let matchIndex = currentMatchIndex = parseInt(nodeIndex, 10);
     if (all || matchIndex === searchState.index) {
       if (text.length) {
-        nodes[i].firstChild.nodeValue = text;
+        nodes[i].innerText = text;
```

Comment:
Yes it should be, because `nodes[i]` is the searchreplace wrapper span and it only ever gets the single text node added here: https://github.com/tinymce/tinymce/blob/236891ffcdf2fd1b698d020e67c31838fb110019/modules/tinymce/src/plugins/searchreplace/main/ts/core/FindMark.ts#L34. Now even if that did have 2 nodes just replacing the first text node would be incorrect because it means you'd end up with the match only being partly replaced.

Reply:
Also happy to swap it back and add the relevant checks to make strict mode compile though if we'd prefer to play it safe.

Reply:
My preference would be for a lower-risk approach in this type of PR. But you're probably right about the current code not really handling all situations.

Reply:
I'm happy with that explanation, merge as is